layout/rowwrap.go

```diff
@@ -0,0 +1,88 @@
+package layout
+
+import (
+	"fyne.io/fyne/v2"
+	"fyne.io/fyne/v2/theme"
+)
+
+type rowWrapLayout struct {
+	rowCount          int
+	horizontalPadding float32
+	verticalPadding   float32
+}
+
+// NewRowWrapLayout returns a layout that dynamically arranges objects
+// with the same height in rows and wraps them as necessary.
+//
+// Object visibility is supported.
+//
+// Since: 2.7
+func NewRowWrapLayout() fyne.Layout {
+	return &rowWrapLayout{
+		horizontalPadding: theme.Padding(),
+		verticalPadding:   theme.Padding(),
+	}
+}
+
+// NewRowWrapLayoutWithCustomPadding creates a new RowWrapLayout instance
+// with the specified paddings.
+//
+// Since: 2.7
+func NewRowWrapLayoutWithCustomPadding(horizontal, vertical float32) fyne.Layout {
+	return &rowWrapLayout{
+		horizontalPadding: horizontal,
+		verticalPadding:   vertical,
+	}
+}
+
+var _ fyne.Layout = (*rowWrapLayout)(nil)
+
+// MinSize finds the smallest size that satisfies all the child objects.
+// For a RowWrapLayout this is the width of the widest child
+// and the height of the first child multiplied by the number of children,
+// with appropriate padding between them.
+func (l *rowWrapLayout) MinSize(objects []fyne.CanvasObject) fyne.Size {
```

Comment:
For something with wrapping like this, (and Label/RichText etc with wrapping), the behavior needed IIRC is for the MinSize returned to depend on the current size of the object. If the object hasn't been Resized (IE its size is 0, 0), then it returns the MinSize as you currently calculate it. But if it has been resized to a width >= the min width, the MinSize height is based on how many rows it needs to lay things out *at the current width*. Otherwise, placing it as the child of another layout will always make it take up more height than needed.

Not sure how to do this with a layout though, given that they are stateless by design. I suppose in Layout you could remeber the size in the `rowWrapLayout` struct, and then reference it in MinSize

Reply:
Layouts don't have to be stateless (see Border), the thing they lack is any connection to the container they are laying out.

Reply:
I spoke too soon as my attempt did not fully work. I have reverted back to the previous solution for now and will make another attempt.

Reply:
@andydotxyz I tried to implement this comment as suggested by dwey. But I could not get it to work and have rolled back those changes completely with the force-push (sorry). All other comments have been implemented though, so this is currently the last open point.

Reply:
@dweymouth I must confess I was a bit confused about your comment at first, because MinSize() was already returning the actual height based on the number of rows as calculated in Layout(). Nevertheless I have now extended it as per your suggestion to return both the actual width and height once it is known (i.e. after Layout() has run).

Reply:
@dweymouth Github is still showing a request for change from you, but I can not find it. Might have to do with my force push earlier. Are you maybe able to close this or is there maybe another open issue that I missed?